File: iot-gateway/src/main/java/NEU/ZuoHao/connecteddevices/labs/common/SensorData.java
package NEU.ZuoHao.connecteddevices.labs.common;

import java.text.SimpleDateFormat;
import java.util.Date;

public class SensorData{
	
	// initialize the timeStamp, name, curValue, avgValue, minValue, maxValue, totValue and sampleCount
	public String name = "Hao's Data";
	public String timeStamp = null;
	public float  curValue = 0.0f;
	public float  avgValue = 0.0f;
	public float  minValue = 0.0f;
	public float  maxValue = 0.0f;
	public float  totValue = 0.0f;
	public int    sampleCount = 0;
	
	// constructor of SensorData class and setup time now
	public SensorData()
	{
		super();
	    updateTimeStamp();
		
	}
	
	//addValue from newVal
	public void addValue(float val)
	{
	    updateTimeStamp();
	    ++sampleCount;
	    curValue  = val;
	    totValue += val;
	    if (curValue < minValue) {
	          minValue = curValue;
	}
	    if (curValue > maxValue) {
	          maxValue = curValue;
	}
	    if (totValue != 0 && sampleCount > 0) {
	          avgValue = totValue / sampleCount;
	} }
	
	//get the average value
	public float getAvgValue()
	{
	    return avgValue;
	}
	
	//get max value
	public float getMaxValue()
	{
		
		return maxValue;
    }
	
	//get minimum value
    public float getMinValue()
    {
        return minValue;
    }
    
    //get the name
    public String getName()
    {
        return name;
    }
    
    //get the value
    public float getValue()
    {
        return curValue;
    }
    
    // set the name
    public void setName(String name)
    {
        if (name != null && name.trim().length() > 0) {
              this.name = name;
} }
   
    // update the time
    public void updateTimeStamp()
    {
        timeStamp = new SimpleDateFormat("yyyy.MM.dd HH:mm.ss").format(new Date());
    }
    
    //set the output format of all data 
    public String output() {
		String customStr = this.name + ":\n" + 
							"\tTime:      " + this.timeStamp + "\n" + 
							"\tCurrent:      " + this.curValue + "\n" +
							"\tAverage:      " + this.avgValue + "\n" +
							"\tSamples:      " + this.sampleCount + "\n" +
							"\tMin:      " + this.minValue + "\n" +
							"\tMax:      " + this.maxValue;
		return customStr;
	}
	
}

File: iot-devices/apps/labs/common/SensorData.py
import os
from datetime import datetime
import json

class SensorData():
    timeStamp = None
    name = 'Temperature' 
    curValue = 0
    avgValue = 0
    minValue = 30
    maxValue = 0
    totValue = 0
    sampleCount = 0
    fjsonstring= 'data'
    tjsonstring= 'data'

    def __init__(self):
        self.timeStamp = str(datetime.now())
    def addValue(self, newVal):
        self.sampleCount += 1
        self.timeStamp  = str(datetime.now())
        self.curValue   = newVal
        self.totValue  += newVal
        if (self.curValue < self.minValue):
            self.minValue = self.curValue
        if (self.curValue > self.maxValue):
            self.maxValue = self.curValue
        if (self.totValue != 0 and self.sampleCount > 0):
            self.avgValue = self.totValue / self.sampleCount
    def getAvgValue(self):
        return self.avgValue
    def getMaxValue(self):
        return self.maxValue
    def getMinValue(self):
        return self.minValue
    def getValue(self):
        return self.curValue
    def setName(self, name):
        self.name = name
    
    def toJson(self,data):
        self.fjsonstring=data
        with open("data_file.json", "w") as write_file:
            json.dump(self.tjsonstring, write_file)
    
    def fromJason(self):
        with open("data_file.json", "r") as read_file:
            self.fjsonstring = json.load(read_file)
        return self.fjsonstring
        
        
    def __str__(self):
        customStr = \
            str('name = '+ self.name + ',' + 
            os.linesep + 'TimeStamp:    ' + self.timeStamp + 
            os.linesep + ' curValue = ' + str(self.curValue) + 
            os.linesep + ' avgValue = ' + str(self.avgValue) + 
            os.linesep + ' minValue = ' + str(self.minValue) + 
            os.linesep + ' maxValue = ' + str(self.maxValue) +
            os.linesep + ' totValue = ' + str(self.totValue) +
            os.linesep + ' SampleCount = ' + str(self.sampleCount) 
            )
            
        return customStr
